### DISCUSSION TITLE
Testing during construction

### DISCUSSION THREAD

**Discussion by Surreal62** [srozum/film_camera_tester#43] [Q&A]
I am in the process of building the tester and is it possible to test functionality during the build process, or does the electronics assembly need to be complete with all circuitry connected?
Reason I ask is if I run the LCD connected to the populated shield board, all I see on the LCD are the two rows of blocks. The LCD is lit.
With a switch panel connected and any sensor, it has the same result - LCD lit with two rows of white blocks.
I have not connected the light unit as it is not complete awaiting a component.
If I connect the LCD directly to the standalone Nano, I get the display showing the tester title text but continually changing menus thereafter.

**Comment by Surreal62** [comment]
Issue resolved.
Unit can be tested at assembly stages 
Fault due to defective Nano

**Reply by User-462** [reply]
I had a similar issue. Bad solder joint.  Good idea to do continuity testing on the main board with no Nano, and no power attached. I tested between the nano socket pins to the end of each connector pin.
Another useful test with unit power on is to measure voltage between ground on the Nano to each sensor output pin. With light/no light, I saw 4.95v or 0 with the newer diodes (positive = on).

**Reply by Surreal62** [reply]
I did all you say as a matter of routine. Assemble and test as one goes along.
A replacement cheapie Nano from Ali cured my issue. Highly likely spending time on the original would see that one working again, but my time to cost of Nano's makes that uneconomical. 
Another Nano I tried resulted in an issue where a few power on/off cycles were needed before the system booted up. 
It is all working very well indeed now and has saved me purchasing £9000+ of tester from Japan!
